lib/model-builder.js

```diff
@@ -161,6 +162,7 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
       assert(ModelBaseClass, 'Base model is not found: ' + baseClass);
     }
   }
+  ModelBaseClass.rank = ModelBaseClass.rank || 0;
 
```

Comment:
adding a new property `rank` right on model instance to reflect the model inheritance rank.
This is not finalized yet. I propose to start at rank `0` if base model defaults to model.js, but rank `1` if a different base model is passed with `options.base` or param `parent` (one extra call to `.extend()`)